samples/NativeLibrary/LoadLibrary.c

```diff
@@ -40,29 +42,49 @@ int main()
         return 0;
     }
 
+//Handle to library
+#ifdef _WIN32
+    HINSTANCE handle = LoadLibraryA(PathToLibrary);
+#else
+    void *handle = dlopen(PathToLibrary, RTLD_LAZY);
+#endif
+
+    void **handlePtr = (void **)&handle;
+
     // Sum two integers
-    int sum = callSumFunc(PathToLibrary, "add", 2, 8);
-    printf("The sum is %d \n", sum);
+    int sum = callSumFunc("add", 2, 8, handlePtr);
+    printf("The sum is %d \n\n", sum);
 
     // Concatenate two strings
-    char *sumstring = callSumStringFunc(PathToLibrary, "sumstring", "ok", "ko");
-    printf("The concatenated string is %s \n", sumstring);
+    char *sumstring = callSumStringFunc("sumstring", "ok", "ko", handlePtr);
+    printf("The concatenated string is %s \n\n", sumstring);
+
+    // Concatenate x strings stored in an array
+    char *strArray[4] = {"tragedy", "of", "the", "wise"};
+    char *mergedString = callMergeStrArray("mergestrings", strArray, 4, handlePtr);
+    printf("The merged string is:\n%s\n\n", mergedString);
 
-    // Free string
+    // Populate a struct
+    typedef struct myStruct
+    {
+        char* name;
+        int value;
+    } t_myStruct;
+    t_myStruct* newStruct = malloc(sizeof(t_myStruct));
+    callPopulateStruct("popstruct",newStruct,handlePtr);
+    puts("---Struct Data---");
+    printf("struct.name:\n%s\n", newStruct->name);
+    printf("struct.value:\n%d\n", newStruct->value);
+
+    // Free strings
     free(sumstring);
+    free(mergedString);
 }
 
-int callSumFunc(char *path, char *funcName, int firstInt, int secondInt)
+int callSumFunc(char *funcName, int firstInt, int secondInt, void **handlePtr)
```

Comment:
I think that the real-world would be more likely structured as:

- The signature of the function exposed to the rest of the program would be just `int callSumFunc(int firstInt, int secondInt)`
- HINSTANCE and the function pointers would be cached in static variables